src/steadystateproblem.cpp

```diff
@@ -148,6 +83,89 @@ void SteadystateProblem::workSteadyStateProblem(Solver *solver, Model *model,
                                     SensitivityOrder::first);
 }
 
+NewtonStatus SteadystateProblem::findSteadyState(Solver *solver,
+                                                 NewtonSolver *newtonSolver,
+                                                 Model *model, int it) {
+    /* First, try to run the Newton solver */
```

Comment:
could we flatten/deduplicate this further by introducing `findSteadyStateNewton` and `findSteadyStateSimulation`  where exception behaviour is modulated by a flag which is only passed on the second call to `findSteadyStateNewton`?

Would also be nice if we wouldn't have to have duplicate code to copy of the number of steps. Could be implemented by a context manager which is either explicitly in the exception or goes out of context when the status is returned, see https://github.com/ICB-DCM/AMICI/blob/ffb5f33001047556a718c504ad17bc81ed3e206d/include/amici/forwardproblem.h as example

Reply:
> context manager

Wouldn't use too many of those. In python they are fine, but in C++ usually we don't want to have side effects in destructors. Especially destructors should never `throw`.

Reply:
I really appreciate the idea of further flattening out the logic. I think that makes it much more transparent.

Reply:
Agreed, much cleaner. Pretty happy with the result, thanks for putting in the effort!

Reply:
Welcome and thanks! :)
As however, we need some extended functionality in this thing (the fully adjoint preequilibration) our large scale project, leaving the code in the 2017-version was not really an option any more... ;)
There will be two more PRs, unfortunately, until then:
1. The fully adjoint preequilibration, first only using the algebraic solution (i.e., solving the linear system of equations), see #1128 .
2. Adding an option to workSteadyStateBackward (then automatically preeq and posteq) to use simulation, including interfacing the cvodes api for the forward simulation quadratures...

However, then preequilibration should have become solid, also for really large scale models with whatever conservation laws...